core/services/workflows/metering/metering.go

```diff
@@ -603,7 +634,37 @@ func (r *Report) SendReceipt(ctx context.Context) error {
 		CreditsConsumed:               r.balance.GetSpent().String(),
 	}
 
-	resp, err := r.client.SubmitWorkflowReceipt(ctx, &req)
+	// Retry logic: max 1 retry with 300ms max wait time
+	maxRetries := 1
+	retryDelay := 150 * time.Millisecond // Half of max wait time for single retry
+
+	var resp *emptypb.Empty
+	var err error
+
+	for attempt := 0; attempt <= maxRetries; attempt++ {
```

Comment:
We only exit the loop via the `break`s, right?
```suggestion
	for attempt := 0; ; attempt++ {
```

Reply:
good catch, thanks: 225a954585c54fd205faf73b93ba1a856a676dbd